### PR TITLE
THOR-1152 Streaming Data Notifications (THOR-1183,1184,1185)

### PR DESCRIPTION
**Pull request by arciisine**
Providing groundwork for streaming data changes:

THOR-1184 - Remove Old RequeryInterval Code from Neon UI 
THOR-1185 - Create Streaming Data Notification Component, currently just a button on the appcomponent
THOR-1183 - Receive Messages in the UI from the Server (UI Task)

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -472,6 +476,7 @@ export class AppComponent implements AfterViewInit, OnInit, OnDestroy {
         this.messageReceiver.subscribe('showVisShortcut', (message) => this.updateShowVisShortcut(message));
         this.messageReceiver.subscribe('showFiltersComponentIcon', (message) => this.updateShowFiltersComponentIcon(message));
         this.messageReceiver.subscribe('toggleGear', (message) => this.updateToggleGear(message));
+        fromEvent(new EventSource('/neon/services/dataset/listen'), 'message').subscribe((time) => this.dataAvailableDashboard());
```
Comment (ThomasSchellenbergNextCentury) [inline src/app/app.component.ts]: Hmm, OK, so, I think the we made an assumption when we wrote the ticket, and part of the task wasn't properly communicated:

The events need to be handled in the Neon Middleware.  Please move the EventSource and the hard-coded path (`/neon/services/dataset/listen`) to a function there, probably in the Connection object, and refer to that function here.  The function can take a custom callback (`dataAvailableDashboard`, or whatever).

I've updated the `lorelei-master` to import the Neon Middleware from its GitHub repo (Eric had accidentally committed it to `master`).

You can make the changes branching off from `master`:  https://github.com/NextCenturyCorporation/neon-framework

Sorry for the confusion.  Please let me know if you have any questions.

Comment (arciisine) [reply src/app/app.component.ts]: Okay, will move

Comment (arciisine) [reply src/app/app.component.ts]: Moved and updated

**Comment by ThomasSchellenbergNextCentury** [comment]
@rartiss55 You'll need to change the `neon-framework` dependency in your `package.json` file to pull from the feature branch until it's merged-in:
```
"neon-framework": "git+https://github.com/NextCenturyCorporation/neon-framework.git#THOR-1183-Receive-Streaming-Messages-In-UI"
```

**Comment by ThomasSchellenbergNextCentury** [comment]
@arciisine I'm getting an error on `npm start`:
```
ERROR in src/app/app.component.ts(194,63): error TS2339: Property 'DATASET_UPDATED' does not exist on type 'typeof channels'.
```
However, I can verify that `DATASET_UPDATED` is in my `channels` in `node_modules/neon-framework/dist/neon.js.`  Any ideas?

**Comment by arciisine** [comment]
> @arciisine I'm getting an error on `npm start`:
> 
> ```
> ERROR in src/app/app.component.ts(194,63): error TS2339: Property 'DATASET_UPDATED' does not exist on type 'typeof channels'.
> ```
> 
> However, I can verify that `DATASET_UPDATED` is in my `channels` in `node_modules/neon-framework/dist/neon.js.` Any ideas?
This is tied to a specific neon-framework version as changes were needed there to facilitate streaming.  I pull the framework repo locally and change the npm dependency to "file://../neon-framework" (assuming it's a sibling folder).

**Comment by rartiss55** [comment]
@arciisine For PRs like this in the future, it'd be good to list out the different things needed to run it in the description (i.e. "please run this against xyz branch in neon-server and update neon-framework dependency to abc").

**Comment by ThomasSchellenbergNextCentury** [comment]
@arciisine The badge (red dot) on the notification icon is always there, even when `updatedData` is false.  What's up with that?

**Comment by ThomasSchellenbergNextCentury** [comment]
So, the `refreshDashboard` function wasn't working the way any sane person would expect it to work (edit: 
 I mean, before you touched anything -- I'm not blaming you!), so I pushed a fix for that.  Now all of the visualizations will requery whenever you hit that notification button.

**Review by rartiss55 — CHANGES_REQUESTED** [review]
Looks good -- I'll second Thomas' comment though about the notification badge. That shouldn't be red if updatedData is false.

**Comment by arciisine** [comment]
> @arciisine For PRs like this in the future, it'd be good to list out the different things needed to run it in the description (i.e. "please run this against xyz branch in neon-server and update neon-framework dependency to abc").

Will do, wasn't sure how to communicate these things, so makes sense.

**Comment by arciisine** [comment]
Yeah, that shouldn't be firing on load. Will fix.

**Review by rartiss55 — APPROVED** [review]

**Pull request merged**